src/util/psbt/raw.rs

```diff
@@ -119,3 +136,62 @@ impl Decodable for Pair {
         })
     }
 }
+
+impl From<u8> for ProprietaryType {
+    fn from(_: u8) -> Self {
+        ProprietaryType
+    }
+}
+
+impl Into<u8> for ProprietaryType {
+    fn into(self) -> u8 {
+        0u8
+    }
+}
+
+impl<Subtype> Encodable for ProprietaryKey<Subtype> where Subtype: Copy + From<u8> + Into<u8> {
+    fn consensus_encode<W: io::Write>(&self, mut e: W) -> Result<usize, encode::Error> {
+        let mut len = self.prefix.consensus_encode(&mut e)? + 1;
+        e.emit_u8(self.subtype.into())?;
+        len += e.write(&self.key)?;
+        Ok(len)
+    }
+}
+
+impl<Subtype> Decodable for ProprietaryKey<Subtype> where Subtype: Copy + From<u8> + Into<u8> {
+    fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, encode::Error> {
+        let VarInt(prefix_size) = VarInt::consensus_decode(&mut d)?;
+
+        let mut prefix = vec![0u8; prefix_size as usize];
```

Comment:
There is a `Decodable` implementation for `Vec<u8>`.. So you can just do `let prefix = Vec::<u8>::consensus_decode(&mut d)?;`.